CRM/Report/Form/Member/Detail.php

```diff
@@ -266,6 +303,80 @@ public function from() {
              LEFT JOIN civicrm_contribution {$this->_aliases['civicrm_contribution']}
                  ON cmp.contribution_id={$this->_aliases['civicrm_contribution']}.id\n";
     }
+    if ($this->isTableSelected('civicrm_contribution_recur')) {
+      $this->_from .= <<<HERESQL
+            LEFT JOIN civicrm_contribution_recur {$this->_aliases['civicrm_contribution_recur']}
+                ON {$this->_aliases['civicrm_membership']}.contribution_recur_id = {$this->_aliases['civicrm_contribution_recur']}.id
+HERESQL;
+    }
+  }
+
+  /**
+   * Override to add handling for autorenew status.
+   */
+  public function storeWhereHavingClauseArray() {
+    parent::storeWhereHavingClauseArray();
+
+    // Handle autorenew status
+    $op = $this->_params['autorenew_status_id_op'] ?? NULL;
+    $value = $this->_params['autorenew_status_id_value'] ?? NULL;
+    $clauseParts = [];
+    switch ($op) {
+      case 'in':
+        if ($value !== NULL && is_array($value) && count($value) > 0) {
+          $regularOptions = implode(', ', array_diff($value, [0, -1]));
+          // None: is null
+          if (in_array(0, $value)) {
+            $clauseParts[] = "{$this->_aliases['civicrm_membership']}.contribution_recur_id IS NULL";
+          }
+          // Ended: not null, end_date in past
+          if (in_array(-1, $value)) {
+            $clauseParts[] = <<<HERESQL
+              {$this->_aliases['civicrm_membership']}.contribution_recur_id IS NOT NULL
+                AND {$this->_aliases['civicrm_contribution_recur']}.end_date < NOW()
+HERESQL;
```

Comment:
@agh1 can we avoid this override & instead override whereClause - you can see from the e.g below it's designed to allow  a field (or more) to be overridden with field-specific logic.


```
  /**
   * Generate where clause.
   *
   * This can be overridden in reports for special treatment of a field
   *
   * @param array $field Field specifications
   * @param string $op Query operator (not an exact match to sql)
   * @param mixed $value
   * @param float $min
   * @param float $max
   *
   * @return null|string
   */
  public function whereClause(&$field, $op, $value, $min, $max) {
    if ($field['name'] === 'contact_id' && $value) {
      $this->contactID = (int) $value;
      $mergedContactIDs = $this->getContactsMergedIntoThisOne($this->contactID);
      $clause = parent::whereClause($field, 'in', array_merge([$this->contactID], $mergedContactIDs), $min, $max);
      return $clause;
    }
  }
```